servant/src/Servant/API/IsSecure.hs

```diff
@@ -0,0 +1,7 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+module Servant.API.IsSecure where
+
+import Data.Typeable
+
+data IsSecure = Secure | NotSecure
```

Comment:
Sure. I'm not done on the PR yet, I still have to push some tests for these new combinators and write decent haddocks, `IsSecure` will be documented and link to wai's `Bool`-returning field accessor.